### PR TITLE
Fixes #122

### LINKED ISSUE #122: Warning on indexes for ORDER BY
In the section for sorting (https://docs.hazelcast.com/imdg/latest-dev/sql/select-statement.html#sorting), there is

```
Note that, for this, you need to add indexes to the map objects to be queried and sorted. See the indexing section for information on adding and configuring indexes.
```

This is completely correct. However, it's a significant limitation and I would suggest putting in a big fat warning box. It's kind of getting lost in the paragraph and without this non-obvious limitation, the ORDER BY won't work.

### PR DESCRIPTION
**Pull request by Serdaro**

### CODE REVIEW

File: docs/modules/sql/pages/select-statement.adoc [hunk 1]
```diff
@@ -21,8 +21,9 @@ Any row that doesn't satisfy the condition is eliminated from the result.
 
 == Sorting
 
-You can use standard the SQL clauses `ORDER BY`, `LIMIT`, and `OFFSET` to sort and limit the result set.
-Note that, for this, you need to add indexes to the map objects to be queried and sorted.
+You can use the standard SQL clauses `ORDER BY`, `LIMIT`, and `OFFSET` to sort and limit the result set.
+
+WARNING: Note that, you must add indexes to the map objects to be queried and sorted.
```
Comment (Holmistr) [inline docs/modules/sql/pages/select-statement.adoc:26]: I think we could be even clearer with an example:


WARNING: Note that, you must add sorted indexes to the map object's fields sorted by. For example, suppose you have a query like `SELECT * FROM persons ORDER BY name ASC`, there's has to be a sorted index on the `name` field like:
```
hazelcast:
  map:
    default:
      indexes:
        - type: SORTED
            attributes:
              - "name"
```

Please, verify with @petrpleshachkov that it has to be a sorted index, not just hash. But in general, I would add a specific example as it's much easier to understand.

**Review by Holmistr — CHANGES_REQUESTED** [review]
I would suggest adding an example to make it really clear.

File: docs/modules/sql/pages/select-statement.adoc (hunk 1 above)
Comment (Holmistr) [reply docs/modules/sql/pages/select-statement.adoc:26]: Preferably, the configuration should be present in XML, YAML and programmatic configuration.

**Comment by Serdaro** [comment]
@Holmistr , please see the latest commit in this PR for the improved WARNING.

**Review by Holmistr — APPROVED** [review]

**Pull request merged**